objects/src/batches/note_tree.rs

```diff
@@ -0,0 +1,46 @@
+use miden_crypto::{
+    hash::rpo::RpoDigest,
+    merkle::{MerkleError, SimpleSmt},
+};
+
+use crate::{notes::NoteMetadata, BATCH_OUTPUT_NOTES_TREE_DEPTH};
+
+/// Wrapper over [SimpleSmt] for batch note tree
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct BatchNoteTree(SimpleSmt<BATCH_OUTPUT_NOTES_TREE_DEPTH>);
+
+impl BatchNoteTree {
```

Comment:
Do we not need any other methods here? For example, iterating over notes in the tree.

Reply:
At the moment we're able to rewrite Miden node to the new wrappers. But I haven't tried to rewrite Miden client, maybe it would need other methods.